firebase-config/src/main/java/com/google/firebase/remoteconfig/internal/ConfigRealtimeHttpClient.java

```diff
@@ -288,9 +305,9 @@ synchronized void beginRealtimeHttpStream() {
         resetRetryParameters();
         ConfigAutoFetch configAutoFetch = startAutoFetch(httpURLConnection);
         configAutoFetch.listenForNotifications();
-      } else {
-        retryHTTPConnection();
+        closeRealtimeHttpStream();
       }
+      retryHTTPConnection();
```

Comment:
Looks like we'll always retry the connection even after closing it above?

Reply:
Yeah essentially. configAutoFetch.listenForNotifications is a blocking call as it listens for messages through the stream. As soon as it moves on from that method it means the stream is closed and no messages are left so we'll try to reopen the http connection. retryHTTPConnection has checks to see if the app is in the right condition to retry(i.e no backoff, no current http connection. )